src/pkg/blob/manager.go

```diff
@@ -0,0 +1,165 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/distribution"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/blob/dao"
+	"github.com/goharbor/harbor/src/pkg/q"
+)
+
+var (
+	// Mgr ...
+	Mgr = NewManager()
+)
+
+// Manager ...
+type Manager interface {
+	// AddToArtifact add blob to artifact
+	AddToArtifact(ctx context.Context, blobDigest, artifactDigest string) (int64, error)
```

Comment:
`AddToArtifact` is confusing as we do have artifact model in Harbor, it sounds like it attachs a blob to a existing artifact object.